marconi-chain-index/src/Marconi/ChainIndex/Experimental/Runner.hs

```diff
@@ -56,27 +62,81 @@ runIndexer
   -> indexer (WithDistance BlockEvent)
   -> IO ()
 runIndexer socketPath networkId _startingPoint traceName indexer = do
-  securityParam <- Utils.toException $ Utils.querySecurityParam @Void networkId socketPath
-  eventQueue <- STM.newTBQueueIO $ fromIntegral securityParam
-  cBox <- Concurrent.newMVar indexer
   c <- Tracing.defaultConfigStdout
-  Async.concurrently_
-    ( withTrace c traceName $ \trace ->
-        let io =
-              withChainSyncEventEpochNoStream
-                socketPath
-                networkId
-                [Core.genesis]
-                (mkEventStream eventQueue . chainSyncEventStreamLogging trace)
-            handleException NoIntersectionFound =
-              Trace.logError trace $
-                Pretty.renderStrict $
-                  Pretty.layoutPretty
-                    Pretty.defaultLayoutOptions
-                    "No intersection found"
-         in io `catch` handleException
-    )
-    (readEvent eventQueue cBox)
+  withTrace c traceName $ \trace -> runChainSyncClientWithRetries trace
+  where
+    secondsToWait :: Int
+    secondsToWait = 30
+
+    runChainSyncClientWithRetries
+      :: Trace IO Text
+      -> IO ()
+    runChainSyncClientWithRetries trace = do
+      catches
+        (runChainSyncClient trace)
+        [ Handler $
+            \case
+              e | isDoesNotExistError e -> handleRunIndexersException trace
+              e -> throwIO e
+        , Handler $
+            \case
+              MuxError (MuxIOException _) _ -> handleRunIndexersException trace
+              e -> throwIO e
+        ]
+
+    handleRunIndexersException
+      :: Trace IO Text
+      -> IO ()
+    handleRunIndexersException trace = do
+      logWarning trace $
+        renderStrict $
+          layoutPretty defaultLayoutOptions $
+            "Could not connect to Cardano node: socket file"
+              <+> pretty socketPath
+              <+> "does not exist. Retrying in"
+              <+> pretty secondsToWait
+              <> "s ..."
+      threadDelay $ secondsToWait * 1_000_000
+      runChainSyncClientWithRetries trace
+
+    runChainSyncClient
+      :: Trace IO Text
+      -> IO ()
+    runChainSyncClient trace = do
+      securityParam <- Utils.toException $ Utils.querySecurityParam @Void networkId socketPath
+      eventQueue <- STM.newTBQueueIO $ fromIntegral securityParam
+      cBox <- Concurrent.newMVar indexer
+      let io =
+            withChainSyncEventEpochNoStream
+              socketPath
+              networkId
+              [Core.genesis]
+              (mkEventStream eventQueue . chainSyncEventStreamLogging trace)
+          handleException NoIntersectionFound =
```

Comment:
Same, `handleException` may be `whenNoIntersectionFound`?

Reply:
Changed it.